core/tx_pool.go

```diff
@@ -335,10 +337,18 @@ func NewTxPool(config TxPoolConfig, chainconfig *params.ChainConfig, chain block
 	pool.chainHeadSub = pool.chain.SubscribeChainHeadEvent(pool.chainHeadCh)
 	pool.wg.Add(1)
 	go pool.loop()
+	go pool.periodicallyCalculateRealTxActivity()
 
 	return pool
 }
 
+func (pool *TxPool) periodicallyCalculateRealTxActivity() {
+	ticker := time.NewTicker(time.Second)
+	for range ticker.C {
```

Comment:
This currently doesn't react properly to shutdown and would theoretically continue forever. 

To do shutdown gracefully we need to handle it explicitly. For this we need a `sync.WaitGroup` and a cancelable context or channels. Usually, I'd prefer a context as it is more generic and cleaner but in the pool it is done via the channel `reorgShutdownCh` which is closed in `loop()` on `case <-pool.chainHeadSub.Err()`. 

1. Introduce select and for loop in `periodicallyCalculateRealTxActivity`
2. Detect shutdown with `case <-pool.chainHeadSub.Err()` or introduce separate shutdown channel for this method and exit function
3. add to waitgroup before goroutine call
5. `defer pool.wg.Done()`
6. `defer ticker.Stop()`

Reply:
fixed [b0b4b72](https://github.com/scroll-tech/go-ethereum/pull/983/commits/b0b4b7268238c2f0ac869b7ee2344a30a97b06e2)